x-pack/plugins/security_solution/public/timelines/components/timeline/esql_tab_content/index.tsx

```diff
@@ -128,9 +128,7 @@ export const DiscoverTabContent: FC<DiscoverTabContentProps> = ({ timelineId })
     if (!savedSearchById) {
       // nothing to restore if savedSearchById is null
       if (status === 'draft') {
-        resetDiscoverAppState().then(() => {
-          setSavedSearchLoaded(true);
-        });
+        setSavedSearchLoaded(true);
```

Comment:
`Suggestion` : On second thoughts, I think this whole `useEffect` can be removed since you are already resetting Discover App State using the saved Search ID.